football-gui-game/src/main/kotlin/com/github/lipinskipawel/network/DirectConnection.kt

```diff
@@ -0,0 +1,53 @@
+package com.github.lipinskipawel.network
+
+import com.github.lipinskipawel.board.engine.Move
+import java.net.Socket
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.function.Consumer
+
+internal class DirectConnection private constructor(private val socket: Socket) : Connection {
```

Comment:
This class is over-complicated to me. If I understand correctly the purpose of this class is to send and receive moves in a turn-based game. If that's true then why are you creating a thread that is running constantly and expect receiving a move even if it isn't an opponent move?

Reply:
Yes, the purpose of this class is to send and receive moves.
I'm submitting the task (class `ReadingDataThroughSocket`) to the thread pool in the static 'of` method.
How can I made it more simpler?
Matter of fact I should refactor this to
- move this `pool` from the `DirectConnection` fields to the `of` method
- create Future<*> filed which will represent given task of reading through socket

What do you think?

Reply:
I will leave it as it is since later on I could use this connection to sending messages.